vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp

```diff
@@ -336,8 +336,59 @@ EXPORT(int, _sceKernelGetMutexInfo, SceUID mutexId, SceKernelMutexInfo *pInfo) {
     return SCE_KERNEL_OK;
 }
 
-EXPORT(int, _sceKernelGetRWLockInfo) {
-    TRACY_FUNC(_sceKernelGetRWLockInfo);
+EXPORT(int, _sceKernelGetRWLockInfo, SceUID rwlockId, SceKernelRWLockInfo *info) {
+    TRACY_FUNC(_sceKernelGetRWLockInfo, rwlockId, info);
+    if (!info)
+        return RET_ERROR(SCE_KERNEL_ERROR_INVALID_ARGUMENT);
+    if (info->size < sizeof(SceKernelRWLockInfo))
+        return RET_ERROR(SCE_KERNEL_ERROR_INVALID_ARGUMENT);
+    const RWLockPtr rwlock = lock_and_find(rwlockId, emuenv.kernel.rwlocks, emuenv.kernel.mutex);
+    if (!rwlock)
+        return RET_ERROR(SCE_KERNEL_ERROR_UNKNOWN_RW_LOCK_ID);
+    const std::lock_guard<std::mutex> rwlock_lock(rwlock->mutex);
+    info->rwLockId = rwlock->uid;
+    strncpy(info->name, rwlock->name, KERNELOBJECT_MAX_NAME_LENGTH + 1);
+    info->attr = rwlock->attr;
+    if (rwlock->state == RWLockState::Unlocked) {
+        info->lockCount = 0;
+        info->writeOwnerId = 0;
+        info->numReadWaitThreads = 0;
+        info->numWriteWaitThreads = 0;
+    } else if (rwlock->state == RWLockState::ReadLocked) {
+        int lock_count = 0;
+        for (auto t : rwlock->owners) {
+            lock_count += t.second;
+        }
+        info->lockCount = lock_count;
+        info->writeOwnerId = 0;
+        info->numReadWaitThreads = 0;
+        info->numWriteWaitThreads = 0;
+        if (rwlock->waiting_threads->size() > 0) {
+            STUBBED("info for rw lock with waiting threads not implemented");
+            LOG_TRACE("state:{}, owners.count:{}, waiting_threads->size:{}", (int)(rwlock->state), rwlock->owners.size(), rwlock->waiting_threads->size());
```

Comment:
Can you remove the LOG_TRACE or make them disabled by default? The STUBBED already tell if this is happening and if it does, it is most likely that this will be called many times per frame, completely filling the log.

Reply:
Done.